### PR TITLE
Fix hive connector error when query is looking for a schema which doe…

### LINKED ISSUE #14746: Hive connector returns an error when query is looking for a schema which does not exist
When glue is used as a metastore for a hive-hadoop2 connector the query:

`select * FROM system.jdbc.tables;` 

is working as expected returning results where the user has an access to. However, when the user adds additional condition like:

`select * FROM system.jdbc.tables where table_schem='public';`

The query fails with following error:

`io.trino.spi.TrinoException: User: arn:aws:sts::<redacted> is not authorized to perform: glue:GetTables on resource: arn:aws:glue:<redacted> because no identity-based policy allows the glue:GetTables action (Service: AWSGlue; Status Code: 400; Error Code: AccessDeniedException; Request ID: <redacted>; Proxy: null)`

The expectation for the second query is to return only a subset of the results form the first query or to return no results instead of the error. This situation happen when the specified schema does not exist. 

Stacktrace of the error is attached.
[stacktrace.txt](https://github.com/trinodb/trino/files/9858681/stacktrace.txt)

### PR DESCRIPTION
**Pull request by albericgenius**
## Description
Fix #14746

## Non-technical explanation
According to the proposal solution, when we don't have access, so we don't know if it exists, assume it does not

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`14746`)
```

### CODE REVIEW

**Comment by findinpath** [comment]
As seen in the stacktrace attached to https://github.com/trinodb/trino/issues/14746 the exception is `AccessDeniedException` which is much more specific than `AmazonServiceException` :

```
Caused by: com.amazonaws.services.glue.model.AccessDeniedException: User: arn:aws:sts::<redacted> is not authorized to perform: glue:GetTables on resource: arn:aws:glue:<redacted> because no identity-based policy allows the glue:GetTables action (Service: AWSGlue; Status Code: 400; Error Code: AccessDeniedException; Request ID: <redacted>; Proxy: null)
```

Note also that the exception in the stacktrace originates from `getAllViews` method - so probably this should be adapted as well:

```
io.trino.plugin.hive.metastore.glue.GlueHiveMetastore.getAllViews(GlueHiveMetastore.java:466)
```

**Comment by findinpath** [comment]
```
Fix hive connector error when query is looking for a schema which does not exist
```

The commit message is actually misleading. This is not what your change does.


What actually happens is that the access denied exception is swallowed for usability sake. I'm not yet convinced that this is the way to go. A user missing the `glue:GetTables` permission may be confused why aren't there any views/tables listed in a specific Trino schema. Does this make sense from a functional point of view?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 1]
```diff
@@ -432,6 +432,10 @@ public List<String> getAllTables(String databaseName)
             return ImmutableList.of();
         }
         catch (AmazonServiceException e) {
+            if (!getAllDatabases().contains(databaseName)) {
```
Comment (alexjo2144) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:435]: I think I would do this differently. Would something more like this work? Add on to the catch above

```
catch (EntityNotFoundException | AccessDeniedException e) {
    return ImmutableList.of();
}
```

**Comment by albericgenius** [comment]
> @albericgenius do you want me to find someone on my team to help with testing this?

Yes, feel free to help with testing this

**Comment by ebyhr** [comment]
You can refer to #14971 for the test example.

**Comment by albericgenius** [comment]
@ebyhr Thanks for knowledge sharing, I created the Test case, but I am not sure it can work, because I need to setup AWS account/permission to pass the test case in my local environment.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveGlueCatalogConnectorSmokeTest.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
+/*
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveGlueCatalogConnectorSmokeTest.java:1]: Please refer to #14998. I don't think we want to add a new smoke test in this PR.

Comment (albericgenius) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveGlueCatalogConnectorSmokeTest.java]: fixed, thanks for help :)

**Comment by findepi** [comment]
> > @albericgenius do you want me to find someone on my team to help with testing this?
> 
> Yes, feel free to help with testing this

@albericgenius thanks.

I think @findinpath @ebyhr will take it from there.
We're still discussing internally how to best test that, but due to how AWS testing resources are managed (for security reasons, they are not open to every contributor), this will be best handled by us.

**Comment by ebyhr** [comment]
Superseded by #14998

**Pull request closed without merging**